### ISSUE TITLE
DatePicker: clear button is not accessible

### ISSUE THREAD

**Issue by Pareder** [ant-design/ant-design#57933]
### Reproduction link

[https://ant.design/components/date-picker#date-picker-demo-basic](https://ant.design/components/date-picker#date-picker-demo-basic)

### Steps to reproduce

1. Focus date picker with `allowClear` prop set to `true`
2. Select any date
3. Press Tab

### What is expected?

The clear button to be focused.

### What is actually happening?

Next interactive element is focused.

`ClearIcon` (https://github.com/react-component/picker/blob/master/src/PickerInput/Selector/Icon.tsx#L31) should either:
- pass `tabIndex={0}` and `onKeyDown` handler
- be refactored to `button` element

| Environment | Info |
| --- | --- |
| antd | 6.3.7 |
| React | 19 |
| System | Windows 11 |
| Browser | Google Chrome latest |

---
The same issue occurs for `RangePicker` and `TimePicker`.

**Comment by afc163** [comment]
Confirmed — this is the same pattern as the Input clear button accessibility issue that was fixed in v6.3.7 (#57432). The fix needs to be applied to `rc-picker`'s `ClearIcon` component to make it keyboard-focusable. Thanks @dosu for the reference.
